iceoryx_hoofs/include/iceoryx_hoofs/cxx/string.hpp

```diff
@@ -77,8 +77,8 @@ using IsCxxStringAndCxxStringOrCharArrayOrChar =
 ///     auto bar = iox::cxx::concatenate(fuu, "ahc");
 /// @endcode
 template <typename T1, typename T2>
-IsCxxStringOrCharArrayOrChar<T1, T2, string<internal::SumCapa<T1, T2>::value>> concatenate(const T1& str1,
-                                                                                           const T2& str2) noexcept;
+auto concatenate(const T1& str1, const T2& str2) noexcept
```

Comment:
I would like to disable the rule about trailing return syntax for template types globally. If we decide to do so, then this change and others are not necessary anymore.

Reply:
Needs to be undone.